andela/ah-the-phoenix
# #163388144 Add User Notifications

**What does this PR do?**
-Sends email notifications to followers whenever an article is created
-Sends In-app notifications to followers whenever an article is created
**Description of Task to be completed?**
-Users can subscribe to both in-app and email notifications
-Users can unsubscribe to both in-app and email notifications
-Followers can receive both in-app and email notifications whenever an article is created
**How should this be manually tested?**
* checkout the branch `ft-create-user-notifications-163388144`
* run migrations `python manage.py migrate`
* after running the server signup two users
* user 1 follow user 2
* user 1 unsubscribe from notifications `api/notifications/unsubscribe`
* User 2 create an article
* User 1 check notifications `api/notifications`
* User 1 should have no notifications
* user 1 subscribe for notifications `api/notifications/subscribe`
* User 2 Create an article
* User 1 should receive notifications
**What are the relevant pivotal tracker stories?**
[#163388144](https://www.pivotaltracker.com/n/projects/2240061) 
**Screenshots to support the PR**
1. Email notification
![image](https://user-images.githubusercontent.com/28762562/53173645-27100180-35f9-11e9-846f-59f44dc9bedd.png)
2.In-app Notification
![image](https://user-images.githubusercontent.com/28762562/53173682-3727e100-35f9-11e9-8502-1902d24cab9e.png)
3. Subscribe to notifications
![image](https://user-images.githubusercontent.com/28762562/53182048-bfae7d80-3609-11e9-99d6-4d920ee0e1dd.png)
4. Unsubscribe to notifications
![image](https://user-images.githubusercontent.com/28762562/53182386-70b51800-360a-11e9-86a9-0aa8abb5b51e.png)
![image](https://user-images.githubusercontent.com/28762562/53182398-76126280-360a-11e9-9749-1e5ff29a95e2.png)
